test/backend/routes.spec.js

```diff
@@ -377,6 +385,53 @@ describe('Authentication - ', () => {
         }).catch(done);
     });
 
+    it('on-premise - unauthorized user is not allowed access', function() {
+        saveSetting('IS_RUNNING_INSIDE_ON_PREM', true);
+
+        return GET('settings').then(res => {
+            assert.equal(res.status, 401);
+        });
+    });
+
+    it('on-premise - not allowed acccess if login failed', function(done) {
+        saveSetting('IS_RUNNING_INSIDE_ON_PREM', true);
+
+        POST('oauth2', {access_token: 'invalid access token'})
+        .then(res => {
+            assert.equal(res.status, 500);
+            return GET('settings').then(res => {
+                assert.equal(res.status, 401);
+                done();
+            }).catch(done);
+        })
```

Comment:
missing a `.catch(done)` here (and you can remove the `.catch(done)` above since you `return` the promise